### PR TITLE
Fixes a case where user deletion does not appear on activity logs

### LINKED ISSUE #3310: Apparently user deletions are not being logged in the activity log
Research if user deletions is not getting logged in the activity log.

### PR DESCRIPTION
**Pull request by grafitto**
fixes #3310 

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [x] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

File: app/api/activitylog/activitylogMiddleware.js [hunk 1]
```diff
@@ -27,8 +27,8 @@ export const BODY_REQUIRED_ENDPOINTS = [
 function mustBeLogged(baseurl, method, body) {
   const isLoggedRequest =
     baseurl.includes('/api/') &&
-    !ignoredMethods.includes(method) &&
-    !IGNORED_ENDPOINTS.includes(baseurl);
+    ((!ignoredMethods.includes(method) && !IGNORED_ENDPOINTS.includes(baseurl)) ||
```
Comment (konzz) [inline app/api/activitylog/activitylogMiddleware.js:30]: I think this solution is too specific for the problem, if we found more scenarios like this we dont want to have a long list of "ifs" here.

Probaly making the ignored endpoints a combination of method + endpoint is better.

File: app/api/activitylog/activitylogMiddleware.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import activitylog from './activitylog';
 
 const ignoredMethods = ['GET', 'OPTIONS', 'HEAD'];
 export const IGNORED_ENDPOINTS = [
-  '/api/users',
+  // '/api/users',
```
Comment (grafitto) [inline app/api/activitylog/activitylogMiddleware.js:7]: I removed this endpoint completely from IGNORED_ENDPOINTS. The reason is, only 3 methods exists on this endpoint, POST, GET and DELETE, POST and DELETE need to be logged and GET should not be logged. Which is taken care of by `ignoredMethods` array... @konzz @mfacar I am not sure why the endpoint was included, kindly shed more light into this. Also thanks @daneryl for the help

File: app/api/activitylog/activitylogMiddleware.js [hunk 3]
```diff
@@ -39,6 +39,7 @@ export default (req, _res, next) => {
   if (mustBeLogged(baseurl, method, body)) {
     const time = Date.now();
     const expireAt = date.addYearsToCurrentDate(1);
+    if (body.password) body.password = '*****';
```
Comment (grafitto) [inline app/api/activitylog/activitylogMiddleware.js:42]: I also removed this coz @daneryl noticed the password is displayed in plain text on the logs

File: app/api/activitylog/activitylogMiddleware.js (hunk 2 above)
Comment (mfacar) [reply app/api/activitylog/activitylogMiddleware.js:7]: @grafitto I think it's fine to remove 'api/users' from the ignored endpoints, I'm not sure when it was added but as you found it is used to update users.

**Review by daneryl — CHANGES_REQUESTED** [review]
the log that appears when POST api/users its marked as RAW, what is the next thing so that this appears properly on the logs ? @RafaPolit @mfacar

File: app/api/activitylog/activitylogMiddleware.js (hunk 2 above)
Comment (daneryl) [reply app/api/activitylog/activitylogMiddleware.js:7]: good, we can remove the commented line then :)

**Comment by mfacar** [comment]
@daneryl, @grafitto, To see the API/users endpoint formatted it is needed to add the entry value in activitylogParser.ts

File: app/api/activitylog/activitylogMiddleware.js (hunk 2 above)
Comment (grafitto) [reply app/api/activitylog/activitylogMiddleware.js]: Done

**Comment by grafitto** [comment]
> @daneryl, @grafitto, To see the API/users endpoint formatted it is needed to add the entry value in activitylogParser.ts

Done, I hope that is how it should be done 😄 @mfacar

**Review by daneryl — APPROVED** [review]

**Pull request merged**